### PR TITLE
Jonmv/async doc v1 take2

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review. 

This does not compile, as tests are broken. I'll start working on that now.

I already thought of one missing thing: Jetty threads don't currently help in sending enqueued operations.

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

**Comment by jonmv** [comment]
PTAL :)

File: vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java [hunk 1]
```diff
@@ -406,18 +409,14 @@ private void enqueueAndDispatch(HttpRequest request, ResponseHandler handler, Su
             return;
         }
         Operation operation = Operation.lazilyParsed(request, handler, operationParser);
-        if (enqueued.get() == 0 && operation.dispatch()) // Bypass queue if it is empty.
+        if (enqueued.get() == 1 && operation.dispatch()) // Bypass queue if it is empty.
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java:412]: `enqueued` might be positive even when the actual queue is empty (when multiple threads executes `enqueueAndDispatch` simultaneously.

File: vespaclient-container-plugin/src/main/resources/configdefinitions/document-operation-executor.def [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+resendDelayMillis     int default=10
 
 # Bound on number of document operations to keep in retry queue — further operations are rejected
 maxThrottled          int default=1000
 
+# Whether to perform dispatch from Jetty threads
+doDispatchWithEnqueue bool default=true
+
+# Number of workers in the resender thread pool
+resenderCount         int default=1
```
Comment (bjorncs) [inline vespaclient-container-plugin/src/main/resources/configdefinitions/document-operation-executor.def:14]: We will probably want to control resender threads from `ContainerThreadpool` later on.

**Review by bjorncs — APPROVED** [review]

File: vespaclient-container-plugin/src/main/resources/configdefinitions/document-operation-executor.def (hunk 2 above)
Comment (jonmv) [reply vespaclient-container-plugin/src/main/resources/configdefinitions/document-operation-executor.def:14]: Yes, this isn't even used now >_< Let's clean up when we decide on how we do it.

**Review by bjorncs — COMMENTED** [review]
👍

**Pull request merged**